activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -381,6 +381,10 @@ def convert_value(value, options = {}) # :doc:
         end
       end
 
+      def as_self(value)
```

Comment:
I'd rather not change `HWIA` for the sake of the subclass. Maybe we can just override `nested_under_indifferent_access` in the subclass? Or to be more explicit about behaviour, we could alias over top of the superclass implementation:

```
def nested_as_self
  self.class.new(self)
end

alias_method :nested_under_indifferent_access, :nested_as_self
```

Reply:
> Maybe we can just override `nested_under_indifferent_access` in the subclass?

Can't do that sadly `nested_under_indifferent_access` is defined on the `Hash` class https://github.com/rails/rails/blob/840f55567191c96bdf4250a339759dbb532d351e/activesupport/lib/active_support/core_ext/hash/indifferent_access.rb#L23
Overriding it inside the subclass will have no effect

Reply:
Ah, my mistake, but I still think `as_self` is an inheritance hack. Can we just override `convert_value` in the subclass instead?